Release/v15

benjag opened the pull request:
Release branch for 15.0.0. Individual features should PR against this branch.

mhewson commented:
approve


[![Approved with PullApprove](https://img.shields.io/badge/default-approved-brightgreen.svg)](https://pullapprove.com/rei/rei-cedar/pull-request/160/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=rei/rei-cedar)

benjag committed:
feat(lead): add static lead

benjag committed:
feat: add fluid text presets and heading subheading composition block

benjag committed:
chore(lead): merge lead work

benjag committed:
feat(fluid): adds fluid preset and heading subheading composition

benjag committed:
update to cqw

benjag committed:
test(lead): update snapshot

benjag committed:
feat(box): wIP of CdrBox

daemoncron committed:
ACS-3918 - v15 modal teleport

benjag committed:
feat(text): add preset text components

benjag committed:
feat(abstract): add abstract

benjag committed:
feat(title): add title

benjag committed:
feat(fluid): add fluid types scales

benjag committed:
Revert adding box

benjag committed:
feat(choreographer): wIP of CdrChoreographer

benjag committed:
Fluid updates

benjag committed:
test(lead): update snapshots

benjag committed:
Fix casing on import path

benjag committed:
refactor(image): refactors CdrImg to use native CSS image properties

Removes multiple props, the media container pattern, and simplifies markup

BREAKING CHANGE: API has drastically changed as well as underlying markup of the component

benjag committed:
chore(deps): update chromedriver

benjag committed:
Merge pull request #156 from rei/img-refactor

Img refactor

benjag committed:
refactor(lead): move block within lead component for now

benjag committed:
build: clean up exports

benjag committed:
fix(fluid): move line height ratios outside of supports query

benjag committed:
test: add basic snapshot tests

benjag committed:
chore: fix test pathing

benjag committed:
Merge pull request #157 from rei/fluid-presets

Fluid presets

benjag committed:
Merge pull request #154 from rei/pr/v15-modal-teleport-ACS-3918

ACS-3918 - v15 modal teleport

benjag committed:
chore: drop preset naming convention

benjag committed:
revert(card): revert card example removal

benjag committed:
cleanup

benjag committed:
refactor(build): uses vite plugin to normalize and compile individual component CSS

Removes cedar-compiled and cedar-full, removes extract-css script and accompanying dependendcies

BREAKING CHANGE: Removes cedar-compiled and cedar-full, they must be replaced with style.css

benjag committed:
Regenerate preset snapshots

benjag committed:
Merge pull request #158 from rei/build/css-chunking

Build/css chunking

benjag committed:
docs(examples): fix a11y failures

benjag committed:
build(deps): update dependencies

benjag committed:
Merge pull request #159 from rei/build/updates

Build/updates

benjag committed:
chore: update gh actions runner

benjag committed:
chore: update browser targets

benjag committed:
feat(picture): adds picture and creates a shareable prop interface for image and picture

benjag committed:
Fix test path import

benjag committed:
Add snapshots

benjag committed:
fix(interface): fix type error in loading prop

benjag committed:
Merge pull request #161 from rei/feat/picture

Feat/picture

benjag committed:
15.0.0-alpha.0

benjag committed:
fix(lead): address design feedback for lead

benjag committed:
fix(lead): remove commented code and update snapshots

benjag committed:
Merge pull request #163 from rei/fix/lead

Fix/lead

benjag committed:
feat(lead): refactor lead to global container, move landing lead to an exported lead comp

benjag committed:
15.0.0-alpha.1

benjag committed:
Merge pull request #164 from rei/feat/lead-global

Feat/lead global

benjag committed:
Merge branch 'release/v15' of github.com:rei/rei-cedar into release/v15

benjag committed:
build(static): all static assets to use cdr prefix for consistency

benjag committed:
Merge pull request #165 from rei/rename-assets

Rename assets

benjag committed:
Fix import paths

benjag committed:
Merge pull request #166 from rei/fix-reset-import

Fix import paths

benjag committed:
feat(lead): adds surface orientation prop and logic, removes vestigial styling

benjag committed:
feat(lead): adds srcset as landing lead prop, renames surface prop

benjag committed:
Merge pull request #167 from rei/lead-surface

Lead surface

benjag committed:
build(deps): update stylelint

benjag committed:
build(deps): update node

benjag committed:
build(deps): update to Vite 5 and Vitest 1

benjag committed:
build(vite): deletes vitest config and moves test configuration to vite config

Also removes CSS module processing from test env

benjag committed:
Adjust components to correct usage of size props and correct snapshots

benjag committed:
fix(icon): fix type error

benjag committed:
style(scss): stylelinting

Commit:
add docgen fixes

benjag committed:
Merge pull request #171 from rei/v15-docgen-fixes

benjag committed:
Merge pull request #170 from rei/dependency-maintenance

Dependency maintenance

benjag committed:
refactor(tokens): update tokens and use new surface token in lead

mhewson committed:
Merge pull request #172 from rei/token-updates

refactor(tokens): update tokens and use new surface token in lead

benjag committed:
build: removes depth from built asset paths introduced in vite 5  and preserves the flat structure

benjag committed:
Merge pull request #173 from rei/fix-asset-paths

build: removes depth from built asset paths introduced in vite 5  and…

benjag committed:
refactor(text): create base text mixin and standardize css in presets

benjag committed:
fix(picture): have picture import img css automatically

benjag committed:
chore: cleanup of comments and checkbox example

benjag committed:
refactor(presets): refactor how presets and text work together

Base text props are now extended and the presets do not import the text component directly

benjag committed:
test(presets): update snapshots to account for preset refactor

benjag committed:
Merge pull request #174 from rei/css-imports

Refactor text presets and clean up issues with CSS importing

benjag committed:
15.0.0-beta.0

benjag committed:
docs(docgen): improve docgen glob, add preview tagging, add tags to image props

benjag committed:
Merge pull request #175 from rei/improve-docgen

docs(docgen): improve docgen glob, add preview tagging, add tags to i…

benjag committed:
feat(input): adds inputContainerClass and repurposes labelClass to be consistent with other comps

benjag committed:
fix(input): add default value to labelClass

benjag committed:
Merge pull request #176 from rei/fix-input-class-props

Fix input class props

benjag committed:
docs(atomics): change description

benjag committed:
15.0.0-beat.1

benjag committed:
refactor(presets): externalize style modules so other components can import them

benjag committed:
refactor(leads): refactors CdrLead in to CdrSplitSurface and moves LandingLead to top-level

benjag committed:
docs(docgen): adds @uses tag for components that use other core components

benjag committed:
test(landinglead): update snapshot to account for baseClass addition

benjag committed:
style(eslint): adjust esline rules

benjag committed:
docs(docgen): replace with CdrIcon

benjag committed:
Merge pull request #177 from rei/refactoring-style-and-lead

Refactoring style and lead

benjag committed:
fix(landinglead): replace object-fit with custom property

benjag committed:
docs(landinglead): reorder usage data

benjag committed:
15.0.0-beta.2

benjag committed:
fix(toast): fix CSS display inheritance

benjag committed:
docs(splitsurface): rewrite component description

benjag committed:
docs(quote): ignore modifier prop for demo

benjag committed:
feat(splitsurface): adds tag prop to allow elements other than div

benjag committed:
feat(tooltip): allow tooltip to generate a unique ID if none is provided

benjag committed:
fix(tooltip): fix typeerror

benjag committed:
docs: copy edits to component profiles

KenjiCrosland commented on src/components/choreographer/CdrChoreographer.vue:
```diff
@@ -0,0 +1,50 @@
+<script setup lang="ts">
```
I didn't realize we were including this in v15?

benjag replied:
Not exported. Leaving it in for further experimentation

mhewson replied:
I think the idea is we are not speaking about it

mhewson commented on public/cdr-reset.css:
```diff
@@ -2,12 +2,7 @@ html {
   -webkit-box-sizing: border-box;
   box-sizing: border-box;
   font-family: sans-serif;
-  font-size: 10px; /* update postcss.config.js pxtorem if this changes */
```
![giphy](https://github.com/rei/rei-cedar/assets/5115458/66b561a1-6523-409c-8ede-2a4e09372d62)

mhewson replied:
also some future thoughts https://www.youtube.com/watch?v=cCAtD_BAHNw

mhewson commented on src/components/abstract/CdrAbstract.vue:
```diff
@@ -0,0 +1,31 @@
+  </component>
+</template>
+
+<style lang="scss" module src="./styles/CdrAbstract.module.scss">
```
will these also be available in our component variables project?

benjag replied:
No, not at this time. We'd have to figure out providing/exporting the fluid scale to that project

mhewson commented on src/components/abstract/styles/CdrAbstract.module.scss:
```diff
@@ -0,0 +1,16 @@
+@import '../../../styles/settings/index.scss';
+@import '../../../styles/settings/fluid.vars.scss';
+
+.cdr-abstract {
```
the tag itself could have a few additional styles I should think

or maybe this is just a global p tag reset 

like 
```
max-width: 75ch;
text-wrap: pretty;
```

mhewson commented on src/components/image/CdrImg.vue:
```diff
@@ -1,145 +1,59 @@
-    default: '',
-  },
-  /**
-   * Aspect ratio of the media container
```
👍

mhewson commented on src/components/image/CdrImg.vue:
```diff
@@ -1,145 +1,59 @@
+  : props.radius
+);
+
+const imageProperties = computed(() => {
```
rad, I love the internal custom props

mhewson commented on src/components/image/styles/CdrImg.module.scss:
```diff
@@ -1,65 +1,10 @@
-  vertical-align: middle;
-  border-style: none;
-
-  /* Style variants
```
amazing

mhewson commented on src/components/kicker/styles/CdrKicker.module.scss:
```diff
@@ -0,0 +1,9 @@
+@import '../../../styles/settings/index.scss';
+@import '../../../styles/settings/fluid.vars.scss';
+
+.cdr-kicker {
```
`text-wrap: nowrap` ?

mhewson commented on src/components/landingLead/styles/CdrHeadingSubheadingBlock.module.scss:
```diff
@@ -0,0 +1,16 @@
+@import '../../../styles/settings/index.scss';
+@import '../../text/presets/styles/CdrHeadingDisplay.module.scss';
+@import '../../text/presets/styles/CdrSubheadingSans.module.scss';
+
```
lets give our headings and titles 
`text-wrap: balance; `

mhewson commented on src/components/modal/CdrModal.vue:
```diff
@@ -314,85 +374,87 @@ onUnmounted(() => {
-        :class="[style['cdr-modal__overlay'], overlayClass]"
-      />
-      <!-- This div (and the one below) is used to avoid a screen reader "keyboard" trap where the
+  <Teleport to="body">
```
do we need a note for why teleport? I mean I guess its just native Vue...

mhewson commented on src/components/picture/__tests__/CdrPicture.spec.js:
```diff
@@ -0,0 +1,23 @@
+describe('CdrPicture', () => {
+  it('matches snapshot', () => {
+    const wrapper = shallowMount(CdrPicture, {
+      props: {
```
thoughts on how palette could change the image source? Atlassian has and interesting take 
https://atlassian.design/components/image/examples#dark-mode

mhewson commented on src/components/text/presets/styles/CdrHeadingDisplay.module.scss:
```diff
@@ -0,0 +1,8 @@
+@import './CdrPresets.module.scss';
+
```
lets add
`text-wrap: balance; `
to our headings

benjag committed:
feat(text): add text-wrapping to reset and max-width to  abstract and body

benjag commented on src/components/text/styles/CdrText.module.scss:
```diff
@@ -2,6 +2,5 @@
 @import './vars/CdrText.vars.scss';
 
 .cdr-text {
-  @include cdr-text-default;
```
vestigial

benjag committed:
15.0.0

mhewson commented on public/cdr-reset.css:
```diff
@@ -38,6 +38,14 @@ body {
 }
 
+h1,h2,h3,h4,h5,h6 {
+  text-wrap: balance;
```
💯

mhewson commented on public/cdr-reset.css:
```diff
@@ -38,6 +38,14 @@ body {
+  text-wrap: balance;
+}
+
+p {
```
🥇
